java/code/src/com/suse/manager/utils/PackageUtils.java

```diff
@@ -41,6 +43,40 @@ public static boolean isTypeDeb(Package pkg) {
         return "deb".equals(getArchTypeLabel(pkg));
     }
 
+    /**
+     * Get the channel architecture as a universal arch string, practically
+     * stripping away any arch type suffixes (e.g. "-deb")
+     * @param arch the package arch
+     * @return an package architecture string
+     */
+    public static String getUniversalArchString(PackageArch arch) {
+        return StringUtils.substringBeforeLast(arch.getLabel(), "-deb");
+    }
+
+    /**
+     * Return an EVR string representation in the format "[epoch:]version-release",
+     * stripping away any dummy release strings (e.g. "-X")
+     *
+     * @param evr the package evr
+     * @return string representation of epoch, version and release
+     */
+    public static String getUniversalEvrString(PackageEvr evr) {
+        StringBuilder builder = new StringBuilder();
+        if (StringUtils.isNumeric(evr.getEpoch())) {
+            builder.append(evr.getEpoch());
+            builder.append(":");
```

Comment:
Given the number of times this method is called, I think that we can gain something in terms of performance by using the `char` representation and concatenating the two calls:

```
            builder.append(evr.getEpoch()).append(':');
```

Reply:
Done